navx/robot.py

```diff
@@ -25,32 +25,34 @@ def robotInit(self):
         # self.navx = navx.AHRS.create_i2c()
 
         # Analog input
-        self.analog = wpilib.AnalogInput(navx.getNavxAnalogInChannel(0))
-
-    def disabled(self):
+        # It seems as though analog is not currently supported. self.analog = wpilib.AnalogInput(navx.pins.getNavxAnalogInChannel(0))
+        
+    def robotPeriodic(self):
 
         self.logger.info("Entered disabled mode")
 
         self.timer.reset()
         self.timer.start()
 
+        
+
         while self.isDisabled():
 
             if self.timer.hasPeriodPassed(0.5):
-                self.sd.putBoolean(
-                    "SupportsDisplacement", self.navx._isDisplacementSupported()
+                wpilib.SmartDashboard.putNumber(
+                    "Displacement X", self.navx.getDisplacementX()
                 )
-                self.sd.putBoolean("IsCalibrating", self.navx.isCalibrating())
-                self.sd.putBoolean("IsConnected", self.navx.isConnected())
-                self.sd.putNumber("Angle", self.navx.getAngle())
-                self.sd.putNumber("Pitch", self.navx.getPitch())
-                self.sd.putNumber("Yaw", self.navx.getYaw())
-                self.sd.putNumber("Roll", self.navx.getRoll())
-                self.sd.putNumber("Analog", self.analog.getVoltage())
-                self.sd.putNumber("Timestamp", self.navx.getLastSensorTimestamp())
-
-            wpilib.Timer.delay(0.010)
-
+                wpilib.SmartDashboard.putNumber(
```

Comment:
Why all the changes from self.sd to wpilib.SmartDashboard? self.sd is probably mildly more performant, and should still work?

Reply:
I could not get sd working initially this year, and as I was troubleshooting, the later stuck.  I will revert the change.  It is good to know.

Reply:
I made the changes.  Self.sd does work on the sim anyway, so I put it back in.  I also ran everything through Black.